### ISSUE TITLE
Timbre interop issue: `spy` vs `spy!`

### ISSUE THREAD

**Issue by lvh** [taoensso/telemere#42]
FYI I tried to port a library that was using timbre via the interop namespace. I ran into an issue that timbre's `spy` doesn't seem to exist in telemere. There is a `spy!`, but it's not obvious that's the same thing. I *think* it is, because there's no equivalent `spy!` in timbre (I think?).

Anyway, I think there should be an additional `spy` macro in the interop namespace, and I think it should probably just be a different spelling of `spy!` so as to not break new or existing callers who've already made the switch.

Or maybe I just misunderstood the API or the intent of the interop ns entirely 😄

**Comment by ptaoussanis** [comment]
@lvh Hi there, thanks for pinging about this - and sorry about the trouble/confusion!

`spy!` should indeed be called `spy`, the "!" suffix was an unintentional mistake. I'll get this fixed for v1 final, expected in January.

In the meantime you could use `timbre/spy!` instead. It's not _identical_ to Timbre's own spy, but it is Telemere's equivalent. The main difference is how it handles errors.

I'll document the difference now, then come back to you with the updated docstring 👍

**Comment by ptaoussanis** [comment]
Here's the new docstring for `telemere.timbre/spy`:

```
Note that for extra flexibility and improved interop with Open Telemetry,
this shim intentionally handles errors (forms that throw) slightly differently
to Timbre's original `spy`:
 
  When the given `form` throws, this shim may create an ADDITIONAL signal of
  `:error` kind and level. The behaviour is equivalent to:
 
  (telemere/spy! level              ; May create 0/1 `:spy`   signals with given    level
    (telemere/catch->error! form))  ; May create 0/1 `:error` signals with `:error` level
 
The additional signal helps to decouple the success and error cases for separate
filtering and/or handling.
```

Hopefully that makes sense?

**Comment by lvh** [comment]
Yup, this is perfect. New behavior makes a ton of sense also. Thank you so much!

**Comment by lvh** [comment]
(I'm closing this since I'm the original reporter and it looks like the issue's resolved, though #42 only refs this issue.)

**Comment by ptaoussanis** [comment]
Cool, you're very welcome! Happy new year :-)
